src/VkBootstrap.h

```diff
@@ -557,23 +563,29 @@ class PhysicalDeviceSelector {
 		std::vector<detail::GenericFeaturesPNextNode> extended_features_chain;
 	};
 
+	detail::Result<std::vector<PhysicalDeviceSelector::PhysicalDeviceDesc>>
+	get_physical_device_descriptions() const;
+
 	// We copy the extension features stored in the selector criteria under the prose of a
 	// "template" to ensure that after fetching everything is compared 1:1 during a match.
 
 	PhysicalDeviceDesc populate_device_details(VkPhysicalDevice phys_device,
 	    std::vector<detail::GenericFeaturesPNextNode> const& src_extended_features_chain) const;
 
+	PhysicalDevice populate_physical_device(PhysicalDeviceDesc const& selected_device_desc) const;
+
 	struct SelectionCriteria {
 		PreferredDeviceType preferred_type = PreferredDeviceType::discrete;
 		bool allow_any_type = true;
+		bool require_device_name = false;
 		bool require_present = true;
 		bool require_dedicated_transfer_queue = false;
 		bool require_dedicated_compute_queue = false;
 		bool require_separate_transfer_queue = false;
 		bool require_separate_compute_queue = false;
 		VkDeviceSize required_mem_size = 0;
 		VkDeviceSize desired_mem_size = 0;
-
+		const char* desired_device_name = "";
```

Comment:
Probably should read `requested_device_name` since its a 'required' value not a desired.